### PR TITLE
setting 100px as default width for buttons

### PR DESCRIPTION
**Pull request by shanbady**
### What are the relevant tickets?

Closes https://github.com/mitodl/hq/issues/4717

### Description (What does it do?)
Sets a min width of 100px for buttons

### How can this be tested?
Checkout this branch and ensure all buttons such as the login button have a minimum width of 100px

### CODE REVIEW

**Comment by jonkafton** [comment]
We probably don't want the min width on action and and icon only buttons (e.g. on the LearningResourceCards), though they could probably use separate treatment from the main Button component.

The action buttons have their own widths, so `minWidth: "auto"` in the `actionStyles` around L305 should fix.

image

vs

image

**Review by ChristopherChudzicki — CHANGES_REQUESTED** [review]
See Jon's comment.

Please also take a look at the relevant storybook pages... e.g., 
- http://localhost:6006/?path=/story/smoot-design-button--action-buttons-showcase
- http://localhost:6006/?path=/story/smoot-design-button--buttons-showcase

just to make sure things look reasonable.

**Comment by shanbady** [comment]
> See Jon's comment.
> 
> Please also take a look at the relevant storybook pages... e.g.,
> 
> * http://localhost:6006/?path=/story/smoot-design-button--action-buttons-showcase
> * http://localhost:6006/?path=/story/smoot-design-button--buttons-showcase
> 
> just to make sure things look reasonable.

thanks. I keep forgetting we have something like this

**Comment by shanbady** [comment]
@jonkafton 

just pushed a fix with your suggestion

File: frontends/ol-components/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -94,6 +94,7 @@ const ButtonStyled = styled.button<ButtonStyleProps>((props) => {
       ":disabled": {
         cursor: "default",
       },
+      minWidth: "100px",
```
Comment (ChristopherChudzicki) [inline frontends/ol-components/src/components/Button/Button.tsx:97]: Our tab component is based on `Button` (it uses tertiary buttons). So this increases the width, of tabs, too.

@steven-hatch Do we want tabs to have min-width 100px? I can see pros/cons

Screenshot 2024-06-27 at 4 24 31 PM

Screenshot 2024-06-27 at 4 24 42 PM
Screenshot 2024-06-27 at 4 25 01 PM

Screenshot 2024-06-27 at 4 25 18 PM

Really the only con is width on mobile screens.

## Separate issue

In the screenshots above, you can see the mobile menu button in top-left corner now has wrong width.

@shanbady I'm sorry this came up... `Button` should not have been used for that in the first place.

But, could you fix it so we don't regress? Probably that menu button should be a "Snowflake". I would just apply the relevant styles to it. See figma: https://www.figma.com/design/Eux3guSenAFVvNHGi1Y9Wm/MIT-Design-System?node-id=3852-51162&m=dev

The code is in `frontends/mit-open/src/page-components/Header/MenuButton.tsx`

**Review by ChristopherChudzicki — CHANGES_REQUESTED** [review]
There's a tab question for @steven-hatch.

@shanbady ... requested change: Could you fix the MenuButton to not use `Button`? Probably should just be a snowflake `styled.button(...)`.

File: frontends/ol-components/src/components/Button/Button.tsx (hunk 1 above)
Comment (shanbady) [reply frontends/ol-components/src/components/Button/Button.tsx:97]: Should be addressed now @ChristopherChudzicki

File: frontends/mit-open/src/page-components/Header/MenuButton.tsx [hunk 2]
```diff
@@ -23,22 +23,32 @@ const MenuButtonInner = styled.div({
...
-})
+  textAlign: "center",
+  display: "inline-flex",
+  justifyContent: "center",
+  alignItems: "center",
+  color: theme.palette.text.primary,
+  transition: `background ${theme.transitions.duration.short}ms`,
+  cursor: "pointer",
+  borderWidth: "1px",
+  borderColor: "currentcolor",
```
Comment (ChristopherChudzicki) [inline frontends/mit-open/src/page-components/Header/MenuButton.tsx:41]: Could you remove these? They do nothing with `borderStyle: "none"`.

*In `<Button />` they do have an effect, but then `variant="text"` set borderStyle to none, so they aren't relevant for that variant.*

**Review by ChristopherChudzicki — APPROVED** [review]
👍 Looks good. I would suggest removing the borderWith and borderColor on the menu button, as noted.

**Pull request merged**